### DISCUSSION TITLE
maven dependency cannot be resolved.

### DISCUSSION THREAD

**Discussion by kxuenvoy** [community/community#53262] [Actions]
### Select Topic Area

Bug

### Body

We suddenly cannot build our project by github action due to the maven dependencies are not resolvable.

#13 56.82 * What went wrong:
#13 56.82 Execution failed for task ':lib:compileKotlin'.
#13 56.82 > Failed to calculate the value of task ':lib:compileKotlin' property 'scriptExtensions$kotlin_gradle_plugin_common'.
#13 56.82    > Could not resolve all dependencies for configuration ':lib:detachedConfiguration3'.
#13 56.82       > Could not resolve org.springframework.boot:spring-boot-dependencies:3.0.2.
#13 56.82         Required by:
#13 56.82             project :lib
#13 56.82          > Could not resolve org.springframework.boot:spring-boot-dependencies:3.0.2.
#13 56.82             > Could not get resource 'https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/3.0.2/spring-boot-dependencies-3.0.2.pom'.
#13 56.82                > Could not GET 'https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/3.0.2/spring-boot-dependencies-3.0.2.pom'.
#13 56.82                   > repo.maven.apache.org
#13 56.82          > Could not resolve org.springframework.boot:spring-boot-dependencies:3.0.2.
#13 56.82             > Could not get resource 'https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/3.0.2/spring-boot-dependencies-3.0.2.pom'.
#13 56.82                > Could not GET 'https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/3.0.2/spring-boot-dependencies-3.0.2.pom'.
#13 56.82                   > repo.maven.apache.org
#

**Comment by raphael5178** [comment]
same error here Unknown host repo.maven.apache.org: Name does not resolve

**Comment by Tocktock** [comment]
me too same error. how to fix it ?? any gradle version cannot resolve maven dependency

**Comment by sukolenvo** [comment]
For anyone having trouble with maven central DNS failing today ..... you can set manually in `hosts` file as a temp fix
`146.75.124.215 repo.maven.apache.org`
(c) Thomas

**Update**:  cdn issue fixed this is not needed anymore

**Reply by fridgecrew-hyunchul** [reply]
- name: Add hosts to /etc/hosts
        run: |
          sudo echo "146.75.124.215 repo.maven.apache.org" | sudo tee -a /etc/hosts

**Reply by kxuenvoy** [reply]
when can the service be recovered? Is there any ETA for this P0 issue?

**Reply by tuteng** [reply]
> ```
>   - name: Add hosts to /etc/hosts
>     run: |
>       sudo echo "146.75.124.215 repo.maven.apache.org" | sudo tee -a /etc/hosts
> ```

This worked for me, thanks

**Reply by cnx-brunomoretto** [reply]
if you are trying to build a docker image add the flag to docker build --add-host=repo.maven.apache.org:146.75.124.215

**Comment by cnx-brunomelo** [comment]
Same error here. No plugin can be resolved.

**Comment by pjy1368** [comment]
![image](https://user-images.githubusercontent.com/56083021/233245171-36e47d47-4a39-4815-8bb2-ead1cfb71aa1.png)
I faced the same error.

**Reply by kwonjael** [reply]
저희도 메이븐... 접속이 안된다고 똑같이 나고 있어요.. ㅠ_ㅠ

**Comment by baitoz** [comment]
Work-around is to specify a repo in pom.xml

Repo also available at: https://repo1.maven.org/maven2/

See https://blog.packagecloud.io/how-does-a-maven-repository-work/ section "Maven Central and the Super pom.xml"

"In addition to your project pom.xml, Maven uses a “Super” pom.xml to inherit some default configuration shared by all Maven installations. This is where the default repository, Maven Central is defined"

Hope that helps.

**Comment by ghost** [comment]
This Worked for me
```
    - name: Change DNS to Cloudflare
      run: /usr/bin/echo -e "nameserver 1.1.1.1\nnameserver 8.8.8.8" | sudo tee /etc/resolv.conf
```

You ***should*** put this step right before maven build command

### Examle
```
    - name: Set up JDK 20
      uses: actions/setup-java@v3
      with:
        java-version: '20'
        distribution: 'temurin'
        cache: maven
    - name: Change DNS to Cloudflare
      run: /usr/bin/echo -e "nameserver 1.1.1.1\nnameserver 8.8.8.8" | sudo tee /etc/resolv.conf
    - name: Build with Maven
      run: mvn -B package --file pom.xml
```

**Comment by max-callstack** [comment]
Seems this issue is resolved now, currently works without any configurations before accessing maven repository.

**Comment by Lugo778** [comment]
> ### Select Topic Area
> 
> Bug
> 
> ### Body
> 
> We suddenly cannot build our project by github action due to the maven dependencies are not resolvable.
> 
> #13 56.82 * What went wrong:
> #13 56.82 Execution failed for task ':lib:compileKotlin'.
> #13 56.82 > Failed to calculate the value of task ':lib:compileKotlin' property 'scriptExtensions$kotlin_gradle_plugin_common'.
> #13 56.82    > Could not resolve all dependencies for configuration ':lib:detachedConfiguration3'.
> #13 56.82       > Could not resolve org.springframework.boot:spring-boot-dependencies:3.0.2.
> #13 56.82         Required by:
> #13 56.82             project :lib
> #13 56.82          > Could not resolve org.springframework.boot:spring-boot-dependencies:3.0.2.
> #13 56.82             > Could not get resource 'https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/3.0.2/spring-boot-dependencies-3.0.2.pom'.
> #13 56.82                > Could not GET 'https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/3.0.2/spring-boot-dependencies-3.0.2.pom'.
> #13 56.82                   > repo.maven.apache.org
> #13 56.82          > Could not resolve org.springframework.boot:spring-boot-dependencies:3.0.2.
> #13 56.82             > Could not get resource 'https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/3.0.2/spring-boot-dependencies-3.0.2.pom'.
> #13 56.82                > Could not GET 'https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/3.0.2/spring-boot-dependencies-3.0.2.pom'.
> #13 56.82                   > repo.maven.apache.org
> #

**Comment by chrisvire** [comment]
I am having this problem much later.  I tried setting the dns server and it didn't work-around the issue.

Execution failed for task ':app-android-common-ffmpeg:extractReleaseAnnotations'.
> Error while evaluating property 'hasAndroidAnnotations' of task ':app-android-common-ffmpeg:extractReleaseAnnotations'.
   > Could not resolve all artifacts for configuration ':app-android-common-ffmpeg:releaseCompileClasspath'.
      > Could not find com.arthenica:mobile-ffmpeg-video:4.4.LTS.
        Required by:
            project :app-android-common-ffmpeg

**Comment by chrisvire** [comment]
Under actions/caches, it says there are no caches. I tried following this post to try to clear it anyway:
https://stackoverflow.com/a/73556714/24251256

**Reply by chrisvire** [reply]
It turns out that mobile-ffmpeg has been removed from Maven.
